### ISSUE TITLE
missing support for simple firewall test and middlebox test

### ISSUE THREAD

**Issue by sandsmark** [m-lab/ndt-server#321]
Noticed that suddenly the firewall and middlebox tests didn't run anymore (using web100clt), and it doesn't seem like they're implemented at all here?

So I assume they broke when the servers started switching over to this implementation. Are there any servers that still run the complete/old implementation?

**Comment by laiyi-ohlsen** [comment]
@stephen-soltesz

**Comment by stephen-soltesz** [comment]
You are correct that the Go implementation of the ndt-server does not implement the ndt5 firewall or middlebox tests. And, we know of no remaining ndt servers supporting the web100 instrumentation or firewall and middlebox tests.

Because of the original protocol's reliance on web100 metrics, and the platform migration to tcpinfo-based kernels, we do not plan to support these in the ndt5 protocol. In particular, during the migration from web100 to tcpinfo based kernels, we observed that:
* most clients were now javascript/browser based, which only run upload and download measurements
* that almost all clients running measurements from modern networks include firewalls now
* the duplex mismatch checks are not generally relevant to modern networks, and the client coordination to measure rwnd or cwnd manipulation by middleboxes is not possible using browser-based clients.

However, we are continuing to exploring new measurements that may be possible with the ndt7 protocol.

Can you say more about the use cases where these measurements were helpful for you?

**Comment by sandsmark** [comment]
Sorry for the late reply, but the additional tests were the only reason I used NDT instead of e. g. dslreports.

I'm not familiar enough with NDT to know for sure which test results in practice are gone, but I assume the congestion, duplex mismatch and firewall detection? Are the buffer size analysis/recommendations gone as well? I don't see them anymore in the command line client at least.

But I understand that the value of maintaining support for these is not worth the cost if there in practice isn't any client that users use that can support them. I downloaded and ran the java one once browsers stopped supporting java applets, but I'm not a significant portion of users. :-)
